sprokit/processes/core/motion_detector_process.cxx

```diff
@@ -0,0 +1,179 @@
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "motion_detector_process.h"
+
+#include <vital/util/wall_timer.h>
+#include <vital/algo/motion_detector.h>
+
+#include <sprokit/processes/kwiver_type_traits.h>
+#include <sprokit/pipeline/process_exception.h>
+
+namespace kwiver {
+
+create_config_trait( algo, std::string, "", "Algorithm configuration subblock" );
+
+//----------------------------------------------------------------
+// Private implementation class
+class motion_detector_process::priv
+{
+public:
+  priv();
+  ~priv();
+
+  vital::algo::motion_detector_sptr m_algo;
+  kwiver::vital::wall_timer m_timer;
+
+}; // end priv class
+
+
+// ==================================================================
+motion_detector_process::
+motion_detector_process( kwiver::vital::config_block_sptr const& config )
+  : process( config ),
+    d( new motion_detector_process::priv )
+{
+  make_ports();
+  make_config();
+}
+
+
+motion_detector_process::
+~motion_detector_process()
+{
+}
+
+
+// ------------------------------------------------------------------
+void
+motion_detector_process::
+_configure()
+{
+  scoped_configure_instrumentation();
+
+  vital::config_block_sptr algo_config = get_config();
+
+  // Check config so it will give run-time diagnostic of config problems
+  if ( ! vital::algo::motion_detector::check_nested_algo_configuration_using_trait( algo, algo_config ) )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Configuration check failed." );
+  }
+
+  vital::algo::motion_detector::set_nested_algo_configuration_using_trait( algo, algo_config, d->m_algo );
+
+  if ( ! d->m_algo )
+  {
+    throw sprokit::invalid_configuration_exception( name(), "Unable to create motion detector algorithm" );
+  }
+}
+
+
+// ------------------------------------------------------------------
+void
+motion_detector_process::
+_step()
+{
+  d->m_timer.start();
+
+  //TODO, handle case where this is optionally provided
+  //auto ts = grab_from_port_using_trait( timestamp );
+
+  auto input = grab_from_port_using_trait( image );
+  kwiver::vital::image_container_sptr result;
+
+  {
+    scoped_step_instrumentation();
+
```

Comment:
Usually this scope would only enclose the algorithm portion. Including the "grab_" calls here can skew results since these calls can block for an indefinite period.

Reply:
Fixed the scoping